packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,323 @@
+      .map((fee) => ({
+        type: fee.type,
+        ...getAssetAndChain(fee.asset),
+        amount: fee.amount.toFixed(),
+      }));
+
+    const showBoost = Boolean(
+      swapDepositChannel?.maxBoostFeeBps && swapDepositChannel.maxBoostFeeBps > 0,
+    );
+    const showCcm = Boolean(swapDepositChannel?.ccmGasBudget || swapDepositChannel?.ccmMessage);
+    const showRefund = swapDepositChannel?.fokRefundAddress;
+
+    const [
+      swapEgressFields,
+      refundEgressFields,
+      estimatedDurationSeconds,
+      srcChainRequiredBlockConfirmations,
+    ] = await Promise.all([
+      getEgressStatusFields(
+        swapEgress,
+        swapEgress?.broadcast,
+        ignoredEgresses,
+        'SWAP',
+        swapEgressTrackerTxRef,
+      ),
+      getEgressStatusFields(
+        refundEgress,
+        refundEgress?.broadcast,
+        ignoredEgresses,
+        'REFUND',
+        refundEgressTrackerTxRef,
+      ),
+      srcAsset && destAsset && estimateSwapDuration({ srcAsset, destAsset }),
+      internalSrcAsset && getRequiredBlockConfirmations(internalSrcAsset),
+    ]);
+
+    const response = {
+      state,
+      swapId: swapRequest?.nativeId.toString(),
+      ...(internalSrcAsset && getAssetAndChain(internalSrcAsset, 'src')),
+      ...(internalDestAsset && getAssetAndChain(internalDestAsset, 'dest')),
+      destAddress: readField(swapRequest, swapDepositChannel, failedSwap, 'destAddress'),
+      depositChannel: {
+        createdAt: swapDepositChannel?.createdAt.valueOf(),
+        brokerCommissionBps: swapDepositChannel?.brokerCommissionBps,
+        depositAddress: swapDepositChannel?.depositAddress,
+        expiryBlock: swapDepositChannel?.srcChainExpiryBlock?.toString(),
+        estimatedExpiryTime: swapDepositChannel?.estimatedExpiryAt?.valueOf(),
+        expectedDepositAmount: swapDepositChannel?.expectedDepositAmount?.toFixed(),
+        isExpired: swapDepositChannel?.isExpired,
+        openedThroughBackend: swapDepositChannel?.openedThroughBackend,
+        affiliateBrokers,
+        fillOrKillParams: swapDepositChannel?.fokMinPriceX128
+          ? {
+              retryDurationBlocks: swapDepositChannel.fokRetryDurationBlocks,
+              refundAddress: swapDepositChannel.fokRefundAddress,
+              minPrice: getPriceFromPriceX128(
+                swapDepositChannel.fokMinPriceX128.toFixed(),
+                swapDepositChannel.srcAsset,
+                swapDepositChannel.destAsset,
+              ),
+            }
+          : undefined,
+        dcaParams: swapDepositChannel?.chunkIntervalBlocks
+          ? {
+              numberOfChunks: swapDepositChannel?.numberOfChunks,
+              chunkIntervalBlocks: swapDepositChannel?.chunkIntervalBlocks,
+            }
+          : undefined,
+      },
+      deposit: {
+        amount:
+          readField(swapRequest, failedSwap, 'depositAmount')?.toFixed() ?? pendingDeposit?.amount,
+        transactionRef: depositTransactionRef,
```

Comment:
its called `transactionRef` but `sentTxRef` below - we should adjust one of them to make it consistent

Reply:
if we split up `swap` into `swap`+`egress` we could call it `transactionRef` on the `egress` object (and the `refund` object)